From a01333c325228785ab89add8f73b91af9b6cba20 Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Sat, 10 Oct 2015 01:16:49 +0530
Subject: [PATCH 18/28] FOR_UPSTREAM [VPG]: drm/i915: Check EQ only if CR
 passed

Link training sequence involves two halfs where clock recovery/CR
is checked first and once it is attained we check for
equilization/EQ. This patch fixes current code and allows call
for equalization check only if CR passed.

Change-Id: I61b94e597b54caf29f7b7fb755bb091aa055b8d3
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9749
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/439691
---
 drivers/gpu/drm/i915/intel_dp.c | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index c01fa5f..140cab5 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -2370,8 +2370,10 @@ static bool intel_vlv_psr_do_exit(struct intel_dp *intel_dp, bool disable)
 
 	/* Fast Link training failed. So do full link training */
 	if (!ret) {
-		intel_dp_start_link_train(intel_dp);
-		intel_dp_complete_link_train(intel_dp);
+		ret = intel_dp_start_link_train(intel_dp);
+		if (ret)
+			intel_dp_complete_link_train(intel_dp);
+
 		intel_dp_stop_link_train(intel_dp);
 	}
 
@@ -2836,8 +2838,10 @@ static void intel_enable_dp(struct intel_encoder *encoder)
 	intel_edp_panel_vdd_on(intel_dp);
 	intel_edp_init_train(intel_dp);
 	intel_dp_sink_dpms(intel_dp, DRM_MODE_DPMS_ON);
-	intel_dp_start_link_train(intel_dp);
-	intel_dp_complete_link_train(intel_dp);
+	if (intel_dp_start_link_train(intel_dp) == false)
+		return;
+	if (intel_dp_complete_link_train(intel_dp) == false)
+		return;
 	intel_dp_stop_link_train(intel_dp);
 }
 
@@ -4553,8 +4557,9 @@ intel_dp_check_link_status(struct intel_dp *intel_dp, bool *perform_full_detect)
 	    (!drm_dp_channel_eq_ok(link_status, intel_dp->lane_count))) {
 		DRM_DEBUG_KMS("%s: channel EQ not ok, retraining\n",
 			      intel_encoder->base.name);
-		intel_dp_start_link_train(intel_dp);
-		intel_dp_complete_link_train(intel_dp);
+		if (intel_dp_start_link_train(intel_dp))
+			intel_dp_complete_link_train(intel_dp);
+
 		intel_dp_stop_link_train(intel_dp);
 	}
 }
-- 
1.9.1

